### PR TITLE
optimized product bloc

### PR DESCRIPTION
**Pull request by vikasprashar99**
Issues Solved in PR

- Optimized Product Bloc

- Getting the response by a single function

### CODE REVIEW

File: lib/screens/productsNew/newUI.dart [hunk 1]
```diff
@@ -329,13 +332,16 @@ class _ProductsUINewState extends State<ProductsUINew> {
                                     });
                                     switch (index) {
                                       case 0:
-                                        apiBloc.add(GetVegetables());
+                                        apiBloc.add(GetProducts(
+                                            type: catArray[index].resname));
                                         break;
```
Comment (Rahul-Vijay) [inline lib/screens/productsNew/newUI.dart:337]: No point in having multiple cases here as index will be able to handle it ?

File: lib/screens/productsNew/newUI.dart [hunk 2]
```diff
@@ -14,22 +14,22 @@ import 'package:mvp/bloc/productsapi_bloc.dart';
...
 class Category {
   final String name;
+  final String resname;
   Color backgroundColor;
   Color textColor;
   final bool hasData;
 
   Category(
       {@required this.name,
+      this.resname,
```
Comment (Rahul-Vijay) [inline lib/screens/productsNew/newUI.dart:32]: this is property should be `required` as Bloc requires it

File: lib/screens/productsNew/newUI.dart [hunk 3]
```diff
@@ -14,22 +14,22 @@ import 'package:mvp/bloc/productsapi_bloc.dart';
...
 import 'package:mvp/models/storeProducts.dart';
 import 'package:mvp/screens/common/cartIcon.dart';
 import 'package:mvp/screens/productsNew/details.dart';
 import 'package:mvp/sizeconfig/sizeconfig.dart';
-// import 'package:provider/provider.dart';
 import 'package:shimmer/shimmer.dart';
 
 class Category {
   final String name;
+  final String resname;
```
Comment (Rahul-Vijay) [inline lib/screens/productsNew/newUI.dart:25]: need better variable name, otherwise cannot understand what exactly it signifies or how it is being used

**Review by Rahul-Vijay — CHANGES_REQUESTED** [review]
review my comments

File: lib/screens/productsNew/newUI.dart (hunk 1 above)
Comment (vikasprashar99) [reply lib/screens/productsNew/newUI.dart:337]: yea will remove them

File: lib/screens/productsNew/newUI.dart (hunk 3 above)
Comment (vikasprashar99) [reply lib/screens/productsNew/newUI.dart:25]: ok I will change it

File: lib/screens/productsNew/newUI.dart (hunk 2 above)
Comment (vikasprashar99) [reply lib/screens/productsNew/newUI.dart:32]: okay

**Comment by vikasprashar99** [comment]
@Rahul-Vijay  Done requested changes

File: lib/screens/productsNew/newUI.dart [hunk 4]
```diff
@@ -22,14 +22,14 @@ import 'package:shimmer/shimmer.dart';
 
 class Category {
   final String name;
-  final String resname;
+  final String categoryname;
```
Comment (Rahul-Vijay) [inline lib/screens/productsNew/newUI.dart:25]: need this in camel case. Make sure you follow it throughout

**Review by Rahul-Vijay — CHANGES_REQUESTED** [review]
review syntax

File: lib/screens/productsNew/newUI.dart (hunk 4 above)
Comment (vikasprashar99) [reply lib/screens/productsNew/newUI.dart]: done

**Review by Rahul-Vijay — APPROVED** [review]
✅

**Pull request merged**